ml-proto/spec/eval.ml

```diff
@@ -171,11 +166,15 @@ let rec eval_expr (c : config) (e : expr) =
     let vs = List.map (fun ev -> some (eval_expr c ev) ev.at) es in
     (import c x) vs
 
-  | CallIndirect (x, e1, es) ->
+  | CallIndirect (ftype, e1, es) ->
     let i = int32 (eval_expr c e1) e1.at in
     let vs = List.map (fun vo -> some (eval_expr c vo) vo.at) es in
-    (* TODO: The conversion to int could overflow. *)
-    eval_func c.instance (table c x (Int32.to_int i @@ e1.at)) vs
+    if i < 0l || i >= Int32.of_int (List.length c.instance.module_.it.table) then
```

Comment:
It would be nice if this could be factored out (into, say, a function named `func_indirect`) analogous to other lookups, and reusing the existing `lookup` function.